### ISSUE TITLE
Line Contents not being shown

### ISSUE THREAD

**Issue by Jaya472** [pyutils/line_profiler#152]
I run my .ipynb file in the OneDrive folder via Anaconda JupyterLab. I am using %lprun line profiler and everything except for the Line Contents are displayed with the following message. What might be the mistake? Please, anyone, suggest. I am struggling for 10 hours just on this one thing!

I realize now maybe it is some version problem but don't know what versions I should have to solve this since I had already uninstalled and reinstalled the line profiler.

My python -V gives 3.9.7
My ipython -V gives 8.2.0
line_profiler (i see two versions in Anaconda environment) 3.5.1 and 3.4.0

```
Timer unit: 1e-07 s

Total time: 0.962042 s

Could not find file C:\Users\my name\AppData\Local\Temp\ipykernel_20788\3519952779.py
Are you sure you are running this program from the same directory
that you ran the profiler from?
Continuing without the function's contents.

Line #      Hits         Time  Per Hit   % Time  Line Contents
==============================================================
     1                                           
     2         1         24.0     24.0      0.0  
     3         6        211.0     35.2      0.0  
```

**Comment by GoingMyWay** [comment]
Same issue, have you figured out the reason?

**Comment by Erotemic** [comment]
Getting a MWE would go a long way towards fixing this.

**Comment by Jaya472** [comment]
@Erotemic Oh ok. I never knew all this stuff. So, I now created a gist as you said and I am sharing the link here. I have created a small example to perform the sum of numbers and used the line profiler and showed the corresponding output. Hope this is enough or else let me know. [https://gist.github.com/Jaya472/10805d3c20dea33908de372eee8d8e68](https://gist.github.com/Jaya472/10805d3c20dea33908de372eee8d8e68)

**Comment by Jaya472** [comment]
And as you said about ipynb file, I created a private repository with the same code as I have put in the gist and this is the link. Hope this is what you were intending. [https://github.com/Jaya472/an-mwe-for-line-profiler-problem/blob/f7d7f8c467f152851ede4f105e0b9d2f3b80ccd8/mwe.ipynb](https://github.com/Jaya472/an-mwe-for-line-profiler-problem/blob/f7d7f8c467f152851ede4f105e0b9d2f3b80ccd8/mwe.ipynb)

**Comment by Jaya472** [comment]
> Same issue, have you figured out the reason?

@GoingMyWay Btw, as a workaround (for now) I have been able to solve the problem by running my code in Google Colab. Because as per the message by the compiler, it is a problem of some directory and installations. So I thought to do on google colab where it will take care of where to install line profiler and where to run the code. And it worked.

@Erotemic But of course, I would like to solve the problem for my personal workstation instead of every time using Colab resources. So please let me know if you need any more information or if you want me to try any resolving steps.

**Comment by Jaya472** [comment]
@Erotemic Yes, that's what I was thinking that maybe it will work on your computer with no problem. Because if it is the directory problem or installation path problem or whatever it is, it would have been specific to my computer. But it is also true both my laptop and computer were giving me the same problem.  Maybe something somewhere changed in the Windows update or pip install paths or even the jupyter notebooks' storage locations that there was some mismatch with the profiler.py unable to find the script on its path and causing the error in question.

Thanks for your help and for now, I will continue on google colab.

**Comment by Erotemic** [comment]
I think it could have to do with this function in `line_profiler.py`: 

```
# This could be in the ipython_extension submodule,
# but it doesn't depend on the IPython module so it's easier to just let it stay here.
def is_ipython_kernel_cell(filename):
    """ Return True if a filename corresponds to a Jupyter Notebook cell
    """
    return (
        filename.startswith('<ipython-input-') or
        filename.startswith(tempfile.gettempdir() + '/ipykernel_') or
        filename.startswith(tempfile.gettempdir() + '/xpython_')
    )
```

Could you insert a breakpoint or a print statement to determine what the filename is and if it is missing a case on your machine?

You can find the location line profiler is installed to via:

```
import line_profiler
print(line_profiler.__file__)
```

Just modify that file in an editor. I'm wondering if it is a backslash thing.

**Comment by amirebrahimi** [comment]
Did anyone fine a workaround to this? Having this issue as well.

**Comment by amirebrahimi** [comment]
After bisecting versions of `ipykernel`, I have found that the problem was introduced in version 6.7.0. I am using `jupyterlab` version 6.5.0 and `line-profiler` version 3.5.1.

I think the change is when long filename support was added to the temp paths:
ipython/ipykernel#838

I'm not sure if the issue is in ipykernel or how line-profiler is using it, but hope this helps others.

**Comment by Erotemic** [comment]
Does this issue still exist in the latest release?
